src/legacy/status_im/chat/models/mentions.cljs

```diff
@@ -100,11 +105,13 @@
              (assoc-in [:chats/mentions chat-id :mentions] state)
              (assoc-in [:chat/inputs-with-mentions chat-id] input-segments))}))
 
+(defonce current-call-id (atom 0))
+
 (rf/defn on-change-text
   {:events [:mention/on-change-text]}
   [{:keys [db]} text]
   (let [chat-id (:current-chat-id db)
-        params  [chat-id text]
+        params  [chat-id text (swap! current-call-id inc) (js/Date.now)]
```

Comment:
@qfrank, event handlers are the pure data layer in the frontend. We shouldn't deref or mutate atoms.

Reply:
The better solution is still simple :) You can store the current call ID in the app-db when `:mention/on-change-text` executes. Then, in the event `:mention/on-change-text-success` simply reads from the db.

The other important change is to use the global interceptor `:now`, this will give you back a unix timestamp in millseconds. The result will be available in the coeffects map. You can destructure like this in the event `{:keys [db now]}`.